[Alerting][Docs] Adding template for documenting alert and action types (#92830)
elastic/kibana: docs/user/alerting/stack-alerts/index-threshold.asciidoc

> The index threshold alert type is designed to run an {es} query over indices, aggregating field values from documents, comparing them to threshold values, and scheduling actions to run when the thresholds are met.

Reviewer: ```suggestion
The index threshold alert type runs an {es} query. It aggregates field values from documents, compares them to threshold values, and schedules actions to run when the thresholds are met.
```

Reviewer: Index threshold
   Create the alert
   Define the conditions
   Add action variables
   Example

[float]
==== Create the alert

Fill in the <<defining-alerts-general-details, alert details>>, then select *Index Threshold*.

> [role="screenshot"]

Reviewer: Is this image needed?

[float]
==== Action variables

> When the index threshold alert condition is met, the following variables are available to use inside each action:

Reviewer: See previous suggestion for intro to this section.

`context.value`:: The value for the alert that met the threshold condition.
`context.conditions`:: A description of the threshold condition. Example: `count greater than 4`

<<defining-alerts-actions-details, Add an action>> to the alert to use these variables.

[float]
==== Example

> In this section, you will use the {kib} <<add-sample-data, weblog sample dataset>> to setup and tune the conditions on an index threshold alert. For this example, we want to detect when any of our top three sites have served more than 420,000 bytes over a 24 hour period.

Reviewer: ```suggestion
In this example, you will use the {kib} <<add-sample-data, sample weblog dataset>> to set up and tune the conditions on an index threshold alert. For this example, you want to detect when any of the top three sites serve more than 420,000 bytes over a 24 hour period.
```

> . From the <<alert-management, Alerts and Actions>>, create a new alert and fill in the <<defining-alerts-general-details, general alert details>>. This alert will be checked every 4 hours, and will only execute actions when the alert status changes. Choose the index threshold alert type. 

Reviewer: How about this for the first step.

1. Open the main menu, then click **Stack Management > Alerts and Actions**
2. Create a new alert that is checked every four hours and executes when the alert status changes.
3. Select the **Index threshold** alert type.
```suggestion
.  Open the main menu, then click **Stack Management > Alerts and Actions**.
.  Create a new alert that is checked every four hours and executes when the alert status changes.
.  Select the **Index threshold** alert type.
```

Reviewer: If you use this, please verify that the formatting is correct.

. From the <<alert-management, Alerts and Actions>>, create a new alert and fill in the <<defining-alerts-general-details, general alert details>>. This alert will be checked every 4 hours, and will only execute actions when the alert status changes. Choose the index threshold alert type. 
+
[role="screenshot"]
image::user/alerting/images/alert-types-index-threshold-select.png[Choosing an index threshold alert type]

Click on each clause to open a control that helps you set the value:

[float]

> ==== Index clause

Reviewer: I suggest making each of these a step.

4. Click **Index**, and set **Indices to query** to **kibana_sample_data_logs.**
5.  Set the **Time field** to **@timestamp.**
6. To detect the number of bytes served during the time window, click **When** and select sum as the aggregation, and bytes as the field to aggregate.

and so on.  If you end up with too many steps, you can break it into sections.

[float]
==== Example

In this example, you will use the {kib} <<add-sample-data, sample weblog dataset>> to set up and tune the conditions on an index threshold alert. For this example, you want to detect when any of the top four sites serve more than 420,000 bytes over a 24 hour period.

.  Open the main menu, then click **Stack Management > Alerts and Actions**.

.  Create a new alert that is checked every four hours and executes actions when the alert status changes.

.  Select the **Index threshold** alert type.
+

> [role="screenshot"]

Reviewer: Move this image under step 2.

We want to alert when any site exceeds 420,000 bytes over a 24 hour period, so we'll set the threshold to 420,000 and use the `is above` comparison. 

. To alert when any of the top four sites exceeds 420,000 bytes over a 24 hour period, select `is above` and enter `420000`.
+
[role="screenshot"]
image::user/alerting/images/alert-types-index-threshold-example-threshold.png[Setting the threshold]

[float]
==== Time window clause

Finally, set the time window to 24 hours to complete the alert configuration.

. Finally, click *For the last*, enter `24` and select `hours` to complete the alert configuration.
+
[role="screenshot"]
image::user/alerting/images/alert-types-index-threshold-example-window.png[Setting the time window]

The preview chart will render showing the 24 hour sum of bytes at 4 hours intervals (the *check every* interval) for the past 120 hours (the last 30 intervals).

. The preview chart will render showing the 24 hour sum of bytes at 4 hours intervals (the *check every* interval) for the past 120 hours (the last 30 intervals).
+
[role="screenshot"]
image::user/alerting/images/alert-types-index-threshold-example-preview.png[Setting the time window]

[float]

> ==== Comparing time windows

Reviewer: Can this be step 11?

11.  Change the time window and observe the effect it has on the chart. Compare ....
